README.md
# advance_image_picker

Flutter plugin for selecting **multiple images** from the Android and iOS image library, **taking
new pictures with the camera**, and **edit** them before using such as rotating, cropping, adding
sticker/filters.

*This is an advanced version of [image_picker](https://pub.dev/packages/image_picker) plugin.*

## Key Features

- Display live camera preview in a widget
- Adjust exposure
- Zoom camera preview
- Capture photo without saving into device library
- Capture with preview size mode & full screen size mode
- Select photos from device library by browsing photo albums
- Preview selected images
- Support button label & text translation
- Easy image editing features, such as rotation, cropping, adding sticker/filters
- Allow user add external image editors for editing selected images.
- Support for displaying object detection & OCR result on selected image view
- Allow setting user's own stickers

## Apps using this package

**1. freemar.vn - Shopping app for Vietnamese**

<a href="https://play.google.com/store/apps/details?id=com.freemar.vn" class="download-btn"><i class="bx bxl-play-store"></i>
Google Play</a>
<a href="https://apps.apple.com/vn/app/freemar/id1530667938?l=vi" class="download-btn"><i class="bx bxl-apple"></i>
App Store</a>

**2. trainghiem.vn - Find places to have fun, find places to experience!**

<a target="_blank" data-animation="fadeInRight" data-delay="1.0s" href="https://play.google.com/store/apps/details?id=com.trainghiem.vn" class="btn" tabindex="0" style="animation-delay: 1s;">
Google Play</a>
<a target="_blank" data-animation="fadeInLeft" data-delay="1.0s" href="https://apps.apple.com/vn/app/trainghiemvn/id1537519143?l=vi" class="btn" tabindex="0" style="animation-delay: 1s;">
App Store</a>

**3. Henoo - Heritage in your pocket (Le patrimoine dans la poche)**

<a href="https://play.google.com/apps/testing/fr.henoo.henooapp" class="download-btn"><i class="bx bxl-play-store"></i>
Google Play</a>
<a href="https://apps.apple.com/fr/app/henoo/id1550979414" class="download-btn"><i class="bx bxl-apple"></i>
App Store</a>

## Demo & Screenshots

<img src="https://raw.githubusercontent.com/weta-vn/freemar_image_picker/master/screenshot/1.png" width="100%"/>
<img src="https://raw.githubusercontent.com/weta-vn/freemar_image_picker/master/screenshot/2.png" width="100%"/>

 ---
[*Youtube Demo Link*](https://youtu.be/pl0S72kd0mo)

## Installation

### Dart

This plugin requires dart version 2.16.1 or later

### iOS

Add these settings to the ios/Runner/Info.plist

~~~~
<key>NSCameraUsageDescription</key>
<string>Can I use the camera please?</string>
<key>NSMicrophoneUsageDescription</key>
<string>Can I use the mic please?</string>
<key>NSPhotoLibraryUsageDescription</key>
<string>App need your agree, can visit your album</string>
~~~~

Modify TOCropViewController version in snapshot (ios/Podfile.lock) to 2.6.1

~~~~
TOCropViewController (2.6.1)
~~~~

### Android

Change the minimum Android sdk version to 21 (or higher), and compile sdk to 31 (or higher) in
your `android/app/build.gradle` file (refer example project).

~~~~
compileSdkVersion 31
minSdkVersion 21
~~~~

Change kotlin version to 1.5.21 or later in your `android/build.gradle` file (refer example project)
.

~~~~
ext.kotlin_version = '1.5.21'
~~~~

Add activity and uses-permissions to your `AndroidManifest.xml`, just like next.

```
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="vn.weta.freemarimagepickerexample">

    <application
        android:name="io.flutter.app.FlutterApplication"
        android:label="freemarimagepicker_example"
        android:requestLegacyExternalStorage="true"
        android:icon="@mipmap/ic_launcher">
        <activity
            android:name="com.yalantis.ucrop.UCropActivity"
            android:screenOrientation="portrait"
            android:theme="@style/Theme.AppCompat.Light.NoActionBar"/>
    </application>
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.CAMERA" />
</manifest>
```

## Usages

Add to pubspec

```
dependencies:
  advance_image_picker: $latest_version
```

Import dart code

```
import 'package:advance_image_picker/advance_image_picker.dart';
```

Setting configs & text translate function

```
// Setup image picker configs (global settings for app)
var configs = ImagePickerConfigs();
configs.appBarTextColor = Colors.black;
configs.stickerFeatureEnabled = false; // ON/OFF features
configs.translateFunc = (name, value) => Intl.message(value, name: name); // Use intl function

// Add plugin's string inside lib/l10n/intl_**.arb of your project to use translation
// For example about english messages, declared inside intl_en.arb
{
  "app_title": "FreeMar - Share your things",
  "image_picker_exit_without_selecting": "Do you want to exit without selecting images?",
  "image_picker_select_images_title": "Selected images count",
  "image_picker_select_images_guide": "You can drag images for sorting list...",
  "image_picker_camera_title": "Camera",
  "image_picker_album_title": "Album",
  "image_picker_preview_title": "Preview",
  "image_picker_confirm_delete": "Do you want to delete this image?",
  "image_picker_confirm_reset_changes": "Do you want to clear all changes for this image?",
  "image_picker_no_images": "No images...",
  "image_picker_image_crop_title": "Image crop",
  "image_picker_image_filter_title": "Image filter",
  "image_picker_image_edit_title": "Image edit",
  "image_picker_select_button_title": "Select",
  "image_picker_image_sticker_title": "Image sticker",
  "image_picker_image_addtext_title": "Image add text",
  "image_picker_image_sticker_guide": "You can click on below icons to add into image, double click to remove it from image",
  "image_picker_edit_text": "Edit text",
  "suitcase": "Suitcase",
  "handbag": "Handbag",
  "bottle": "Bottle",
  "tie": "Tie",
  ...
}

// Or you can simply write your own translate function like this
// configs.translateFunc = (name, value) => Intl.message(value, name: name);
   configs.translateFunc = (name, value) {
     switch (name) {
       case 'image_picker_select_images_title':
         return 'Le immagini selezionate contano';
       case 'image_picker_select_images_guide':
         return 'È possibile trascinare le immagini per lelenco di ordinamento...';
       case 'image_picker_camera_title':
         return 'Telecamera';
       case 'image_picker_album_title':
         return 'Album';
       // ... Declare all your strings here
       default:
         return value;
     }
   }
```

Sample for adding external image editors.  
(You can replace ImageEdit widget by your owned image editor widget, that output file after editing
finised)

```
configs.externalImageEditors['external_image_editor_1'] = EditorParams(
      title: 'external_image_editor_1',
      icon: Icons.edit_rounded,
      onEditorEvent: (
              {required BuildContext context,
              required File file,
              required String title,
              int maxWidth = 1080,
              int maxHeight = 1920,
              int compressQuality = 90,
              ImagePickerConfigs? configs}) async => await Navigator.of(context).push(MaterialPageRoute<File>(
              fullscreenDialog: true,
              builder: (context) => ImageEdit(file: file, title: title, maxWidth: maxWidth, maxHeight: maxHeight, configs: configs)))
  );
```

Sample for usage

```
// Get max 5 images
List<ImageObject> objects =
    await Navigator.of(context).push(
        PageRouteBuilder(pageBuilder:
            (context, animation, __) {
  return ImagePicker(maxCount: 5);
}));

if (objects.length > 0) {
  setState(() {
    imageFiles.addAll(objects
        .map((e) => e.modifiedPath)
        .toList());
  });
}
```

## Credits

This software uses the following open source packages:

- camera
- photo_manager
- photo_view
- image
- image_cropper
- image_editor
- flutter_native_image
- path_provider

## Support

If this plugin was useful to you, helped you to deliver your app, saved you a lot of time, or you
just want to support the project, I would be very grateful if you buy me a cup of coffee.

<a href="https://www.buymeacoffee.com/wetavn" target="_blank" rel="ugc"><img src="https://www.buymeacoffee.com/assets/img/custom_images/purple_img.png" alt="Buy Me A Coffee"></a>



CHANGELOG.md
## [0.1.7+1]

* **Update:**
    * Fix required dart version

## [0.1.7]

* **Update:**
    * Upgrade packages
    * Support for flutter 3.x

## [0.1.6+1]

* **Update:**
    * Update readme

## [0.1.6]

* **Update:**
    * Camera, photo manager packages

## [0.1.5+5]

* **Fixbugs:**
    * Check permission for camera & photo gallery
    * Fixbug about OCR text editting

## [0.1.5+1]

* **New features:**
    * Add custom stickers (by @matthewmizzi suggestion)

## [0.1.5]

* **New features:**
    * Object detection & OCR extraction interface for image

## [0.1.4]d

* **New `ImagePickerConfigs` features:**
    * Camera flash mode `flashMode`, the starting flash mode for the camera can be set to other
      values than `FlashMode.auto`. The default is
      `FlashMode.auto` as in previous version where it could not be modified.
    * Option to remove the flash mode toggle button. If you do not want to allow users to change the
      flash mode, you can set `showFlashMode` to false. It defaults to true. If `showFlashMode` is
      false, the flash mode will use what is set by `flashMode`, since user cannot toggle it.
    * Config option to show and hide the lens direction icon button. If you want to show only one
      camera, you may also want to hide the button than enables users switch camera. You can then
      set
      `showLensDirection` to false. If you show just one `cameraLensDirection` and
      `showLensDirection` is true, then the lens direction button is still shown, but disabled as in
      previous versions.
    * Only add the text line for `textSelectedImagesGuide` if it is not empty. If no drag reorder
      guidance label is specified in translation by returning empty string ''
      for `textSelectedImagesGuide`, then the extra line that holds the text will also be removed,
      resulting in more camera view space.
    * For label `textSelectedImagesTitle` only show it, the colon and space after it, if it is not
      empty. If you return a translated string that is empty '', you can get only the selected "
      image / max" count shown, with no label and colon.
    * Use an `IconButton` as done button, instead of the default `OutlinedButton`.
      Set `doneButtonStyle` to `[DoneButtonStyle.iconButton]` for this option. Defaults
      to `DoneButtonStyle.outlinedButton`, that is the same as only option in previous versions. You
      can also change the button icon by defining custom IconData for the `doneButtonIcon`.
    * Hide done action button when no images have been selected. This is optional new behavior that
      can be used instead of default, that disables it. Select the style
      with `doneButtonDisabledBehavior`.
    * Optional alert when removing selected images. If you set
      `showRemoveImageAlert` to false, there is no alert dialog to confirm the remove/delete of an
      image, when user clicks on the delete icon to remove photos and images from the list of images
      to be used, they are just removed immediately. Defaults to true, showing the alert dialog,
      which is same behavior as in previous version.

* Update all dependencies to their latest versions, most notably image_editor to version 1.0.1 that
  now uses Android embedding V2.
* Clean up remaining lint warnings.

## [0.1.3]

* Improve code lintings.

## [0.1.2]

* Add feature to allow user add their owned image editors with ease (from mr.minhthuanit)

## [0.1.1]

* Optimize code by applying lint option from mr.rydmike
* Update camera plugin

## [0.1.0]

* Rewrite image sticker feature
* Update camera plugin to v0.9.1
* Improve UI customization

## [0.0.5+3]

* Fixbug repo

## [0.0.5+2]

* Fixbug UI

## [0.0.5+1]

* Fixbug CameraPreview error while get camera permission

## [0.0.5]

* Fixbug when removing selected images from list
* Fixbug when capture image rotated 270 degree

## [0.0.4+1]

* Update setting instructions for android

## [0.0.4]

* Fixbug image rotated in some android device
* Modified max width / max height default setting for capture images

## [0.0.3]

* Add ON/OFF camera/album feature
* Add lens direction setting feature

## [0.0.2+2]

* Fixbug camera number.

## [0.0.2+1]

* Lock capture orientation.

## [0.0.2]

* Migrated to null safety.

## [0.0.2-nullsafety.1]

* Migrate to null safety.

## [0.0.1+1]

* Format code.

## [0.0.1]

* Initial release.


lib/configs/image_picker_configs.dart
import 'dart:core';

import 'package:camera/camera.dart';
import 'package:flutter/material.dart';

import '../models/image_object.dart';
import '../widgets/editors/editor_params.dart';

export 'package:camera/camera.dart' show FlashMode;

/// Enum used to define the type of used done button by the image picker.
enum DoneButtonStyle {
  /// Use an [OutlinedButton].
  outlinedButton,

  /// Use an [IconButton].
  iconButton,
}

/// Enum used to define the type of of behavior the done button has when
/// no images have been selected that will be returned.
enum DoneButtonDisabledBehavior {
  /// Done button is disabled
  disabled,

  /// Done button is hidden and not shown at all.
  hidden,
}

/// Global configuration for flutter app using advance_image_picker plugin
/// Call once inside application before using image picker functions
///
/// Sample usage
/// Calling in build function of app widget at main.dart
/// ```dart
/// var configs = ImagePickerConfigs();
/// configs.appBarTextColor = Colors.black;
/// configs.translateFunc = (name, value) => Intl.message(value, name: name);
/// ```
class ImagePickerConfigs {
  /// The default constructor is a factory that returns the configuration
  /// singleton of the picker configuration.
  ///
  /// Global configuration for flutter app using advance_image_picker plugin
  /// Call once inside application before using image picker functions
  ///
  /// Sample usage
  /// Calling in build function of app widget at main.dart
  /// ```dart
  /// var configs = ImagePickerConfigs();
  /// configs.appBarTextColor = Colors.black;
  /// configs.translateFunc = (name, value) => Intl.message(value, name: name);
  /// ```
  factory ImagePickerConfigs() {
    return _singleton;
  }

  ImagePickerConfigs._internal();

  /// Singleton object for holding the image picker configuration settings.
  static final ImagePickerConfigs _singleton = ImagePickerConfigs._internal();

  /// UI labels translated function with 2 parameters `name` and `defaultValue`
  /// Declare `name` for what label needs to be translated in localization file,
  /// such as image_picker_select_images_title. Confirm "UI label strings
  /// (for localization)" section below for understanding usage.
  ///
  /// Sample usage:
  ///
  /// If using Intl, function like this:
  /// configs.translateFunc = (name, value) => Intl.message(value, name: name);
  /// If using GetX, function like this:
  /// configs.translateFunc = (name, value) => name.tr;
  late String Function(String, String) translateFunc;

  /// Grid count for photo album grid view.
  ///
  /// Defaults to 4.
  int albumGridCount = 4;

  /// Thumbnail image width.
  ///
  /// Defaults to 80.
  int thumbWidth = 80;

  /// Thumbnail image height.
  ///
  /// Defaults to 80.
  int thumbHeight = 80;

  /// Thumbnail image width inside album grid view.
  ///
  /// Defaults to 200.
  int albumThumbWidth = 200;

  /// Thumbnail image height inside album grid view.
  ///
  /// Defaults to 200.
  int albumThumbHeight = 200;

  /// Max width for output.
  ///
  /// Defaults to 1080.
  int maxWidth = 1080;

  /// Max height for output.
  ///
  /// Defaults to 1920.
  int maxHeight = 1920;

  /// Quality for output.
  ///
  /// Defaults to 90%.
  int compressQuality = 90;

  /// Resolution setting for camera, such as high, max, medium, low.
  ///
  /// Defaults to [ResolutionPreset.high].
  ResolutionPreset resolutionPreset = ResolutionPreset.high;

  /// Enable this option allow image pre-processing, such as cropping,
  /// ... after inputting
  ///
  /// Defaults to true.
  bool imagePreProcessingEnabled = true;

  /// Enable this option to allow image pre-processing, such as cropping,
  /// after editing.
  ///
  /// Defaults to true.
  bool imagePreProcessingBeforeEditingEnabled = true;

  /// Show delete button on selected list.
  ///
  /// Defaults to true.
  bool showDeleteButtonOnSelectedList = true;

  /// Show confirm alert if removing an already selected image.
  ///
  /// Defaults to true.
  bool showRemoveImageAlert = true;

  /// Show confirm alert if exiting with selected image.
  ///
  /// Defaults to true.
  bool showNonSelectedAlert = true;

  /// Enable image crop/rotation/scale function.
  ///
  /// Defaults to true.
  bool cropFeatureEnabled = true;

  /// Enable image filter function.
  ///
  /// Defaults to true.
  bool filterFeatureEnabled = true;

  /// Enable image adjusting function.
  ///
  /// Defaults to true.
  bool adjustFeatureEnabled = true;

  /// Enable sticker adding function.
  ///
  /// Defaults to true.
  bool stickerFeatureEnabled = true;

  // Picker mode settings.

  /// Enable camera as image source.
  ///
  /// Defaults to true.
  bool cameraPickerModeEnabled = true;

  /// Enable device image album as image source.
  ///
  /// Defaults to true.
  bool albumPickerModeEnabled = true;

  /// Detect labels from image function
  Future<List<DetectObject>> Function(String path)? labelDetectFunc;

  /// Max count for label detection
  int labelDetectMaxCount = 5;

  /// Threshold for label detection
  double labelDetectThreshold = 0.7;

  /// Detect OCR from image function
  Future<String> Function(String path, {bool? isCloudService})? ocrExtractFunc;

  /// Custom sticker only flag
  ///
  /// Defaults to false.
  bool customStickerOnly = false;

  /// Custom sticker paths
  List<String> customStickers = [];

  /// Camera direction setting.
  ///
  /// Options:
  ///
  /// * null: use all available camera (default)
  /// * 0: only use front camera
  /// * 1: only use back camera
  int? cameraLensDirection;

  /// Show the lens direction toggle icon button.
  ///
  /// If you want to show only one camera, you may also want to hide the
  /// button than enables users switch camera, then set [showLensDirection]
  /// to false.
  ///
  /// If you show just one [cameraLensDirection] and [showLensDirection] is
  /// true, then the lens direction button is still shown, but disabled.
  ///
  /// Defaults to true.
  bool showLensDirection = true;

  /// Set the default flash mode.
  ///
  /// Options:
  /// * off: Do not use the flash when taking a picture.
  /// * auto: Device decide whether to flash the camera when taking a picture.
  /// * always: Always use the flash when taking a picture.
  /// * torch: In this app treated the same as using always.
  ///
  /// Defaults to [FlashMode.auto].
  FlashMode flashMode = FlashMode.auto;

  /// Show the flash mode icon button.
  ///
  /// If you want to set the FlashMode to a certain mode, typically
  /// [FlashMode.off], and also hide the button than enables users to
  /// change it, then set [showFlashMode] to false.
  ///
  /// Defaults to true.
  bool showFlashMode = true;

  // UI style settings.

  Widget cameraBadQualityAlertDialog = AlertDialog(semanticLabel: "Error occupied");

  /// Default font for the image picker
  String? fontFamily;

  /// Background color of the camera and image picker.
  ///
  /// Defaults to [Colors.black].
  Color backgroundColor = Colors.black;

  /// Background color of the bottom section of the camera.
  ///
  /// Defaults to [Colors.black].
  Color bottomPanelColor = Colors.black;

  /// Background color of the bottom section of the camera when it is used
  /// in full screen mode.
  ///
  /// Defaults to [Colors.black] with 30% opacity.
  Color bottomPanelColorInFullscreen = Colors.black.withOpacity(0.3);

  /// The background color of the [AppBar] in the image picker.
  ///
  /// Defaults to null.
  /// This results in an AppBar background color that follows current theme.
  Color? appBarBackgroundColor;

  /// The text or foreground color of the [AppBar] in the image picker.
  ///
  /// Defaults to null.
  /// This results in an AppBar text color that follows current theme.
  Color? appBarTextColor;

  /// The background color of the image selection completed button.
  ///
  /// This color only applies to the [doneButtonStyle] of style
  /// [DoneButtonStyle.outlinedButton].
  ///
  /// Defaults to null.
  /// This results in [appBarBackgroundColor] being used.
  Color? appBarDoneButtonColor;

  /// The type of button used on the image picker to select images and close
  /// the image picker.
  ///
  /// The default is [DoneButtonStyle.outlinedButton].
  ///
  /// The alternate style [DoneButtonStyle.iconButton] uses an [IconButton] that
  /// is typically used in [AppBar] actions.
  DoneButtonStyle doneButtonStyle = DoneButtonStyle.outlinedButton;

  /// IconData used by the done button when [doneButtonStyle] is
  /// [DoneButtonStyle.iconButton].
  ///
  /// Defaults to Icon.check.
  IconData doneButtonIcon = Icons.check;

  /// Used to define the type of of behavior the done button has when
  /// no images have been selected that will be returned.
  ///
  /// Defaults to [DoneButtonDisabledBehavior.disabled].
  DoneButtonDisabledBehavior doneButtonDisabledBehavior =
      DoneButtonDisabledBehavior.disabled;

  /// Allow add custom image editors from external call.
  ///
  /// Sample usage:
  ///
  /// configs.externalImageEditors['external_image_editor'] = EditorParams(
  ///   title: 'external_image_editor',
  ///   icon: Icons.wb_sunny_outlined,
  ///   onEditorEvent: (
  ///      {required File file,
  ///       required String title,
  ///       int maxWidth = 1080,
  ///       int maxHeight = 1920,
  ///       ImagePickerConfigs? configs}) async => await
  ///         Navigator.of(context).push(MaterialPageRoute<File>(
  ///       fullscreenDialog: true,
  ///       builder: (context) => ImageEdit(file: file, title: title,
  ///         maxWidth: maxWidth, maxHeight: maxHeight, configs: _configs)))
  Map<String, EditorParams> externalImageEditors = {};

  // UI label strings (for localization)

  /// Get localized text for label "image_picker_select_images_title".
  ///
  /// Defaults to "Selected images count".
  String get textSelectedImagesTitle => getTranslatedString(
      "image_picker_select_images_title", "Selected images count");

  /// Get localized text for label "image_picker_select_images_guide".
  ///
  /// Defaults to "You can drag images for sorting list...".
  String get textSelectedImagesGuide => getTranslatedString(
      "image_picker_select_images_guide",
      "You can drag images for sorting list...");

  /// Get localized text for label "image_picker_camera_title".
  ///
  /// Defaults to "Camera".
  String get textCameraTitle =>
      getTranslatedString("image_picker_camera_title", "Camera");

  /// Get localized text for label "image_picker_album_title".
  ///
  /// Defaults to "Album".
  String get textAlbumTitle =>
      getTranslatedString("image_picker_album_title", "Album");

  /// Get localized text for label "image_picker_preview_title".
  ///
  /// Defaults to "Preview".
  String get textPreviewTitle =>
      getTranslatedString("image_picker_preview_title", "Preview");

  /// Get localized text for label "image_picker_confirm".
  ///
  /// Defaults to "Confirm".
  String get textConfirm =>
      getTranslatedString("image_picker_confirm", "Confirm");

  /// Get localized text for label "image_picker_exit_without_selecting".
  ///
  /// Defaults to "Do you want to exit without selecting images?".
  String get textConfirmExitWithoutSelectingImages => translateFunc(
      "image_picker_exit_without_selecting",
      "Do you want to exit without selecting images?");

  /// Get localized text for label "image_picker_confirm_delete".
  ///
  /// Defaults to "Do you want to delete this image?".
  String get textConfirmDelete => getTranslatedString(
      "image_picker_confirm_delete", "Do you want to delete this image?");

  /// Get localized text for label "image_picker_confirm_reset_changes".
  ///
  /// Defaults to "Do you want to clear all changes for this image?".
  String get textConfirmResetChanges => getTranslatedString(
      "image_picker_confirm_reset_changes",
      "Do you want to clear all changes for this image?");

  /// Get localized text for label "yes".
  ///
  /// Defaults to "Yes".
  String get textYes => getTranslatedString("yes", "Yes");

  /// Get localized text for label "no".
  ///
  /// Defaults to "No".
  String get textNo => getTranslatedString("no", "No");

  /// Get localized text for label "save".
  ///
  /// Defaults to "Save".
  String get textSave => getTranslatedString("save", "Save");

  /// Get localized text for label "clear".
  ///
  /// Defaults to "Clear".
  String get textClear => getTranslatedString("clear", "Clear");

  /// Get localized text for label "image_picker_edit_text".
  ///
  /// Defaults to "Edit text".
  String get textEditText =>
      getTranslatedString("image_picker_edit_text", "Edit text");

  /// Get localized text for label "image_picker_no_images".
  ///
  /// Defaults to "No images ...".
  String get textNoImages =>
      getTranslatedString("image_picker_no_images", "No images ...");

  /// Get localized text for label "image_picker_image_crop_title".
  ///
  /// Defaults to "Image crop".
  String get textImageCropTitle =>
      getTranslatedString("image_picker_image_crop_title", "Image crop");

  /// Get localized text for label "image_picker_image_filter_title".
  ///
  /// Defaults to "Image filter".
  String get textImageFilterTitle =>
      getTranslatedString("image_picker_image_filter_title", "Image filter");

  /// Get localized text for label "image_picker_image_edit_title".
  ///
  /// Defaults to "Image edit".
  String get textImageEditTitle =>
      getTranslatedString("image_picker_image_edit_title", "Image edit");

  /// Get localized text for label "image_picker_image_sticker_title".
  ///
  /// Defaults to "Image sticker".
  String get textImageStickerTitle =>
      getTranslatedString("image_picker_image_sticker_title", "Image sticker");

  /// Get localized text for label "image_picker_image_addtext_title".
  ///
  /// Defaults to "Image add text".
  String get textImageAddTextTitle =>
      getTranslatedString("image_picker_image_addtext_title", "Image add text");

  /// Get localized text for label "image_picker_select_button_title".
  ///
  /// Defaults to "Select".
  String get textSelectButtonTitle =>
      getTranslatedString("image_picker_select_button_title", "Select");

  /// Get localized text for label "image_picker_image_sticker_guide".
  ///
  /// Defaults to "You can click on below icons to add into image, double
  /// click to remove it from image".
  String get textImageStickerGuide => getTranslatedString(
      "image_picker_image_sticker_guide",
      "You can click on sticker icons to scale it or double click to "
          "remove it from image");

  /// Get localized text for label "image_picker_exposure_title".
  ///
  /// Defaults to "Exposure".
  String get textExposure =>
      getTranslatedString("image_picker_exposure_title", "Exposure");

  /// Get localized text for label "image_picker_exposure_locked_title".
  ///
  /// Defaults to "Locked".
  String get textExposureLocked =>
      getTranslatedString("image_picker_exposure_locked_title", "Locked");

  /// Get localized text for label "image_picker_exposure_auto_title".
  ///
  /// Defaults to "auto".
  String get textExposureAuto =>
      getTranslatedString("image_picker_exposure_auto_title", "auto");

  /// Get localized text for label "image_picker_image_edit_contrast".
  ///
  /// Defaults to "contrast".
  String get textContrast =>
      getTranslatedString("image_picker_image_edit_contrast", "contrast");

  /// Get localized text for label "image_picker_image_edit_brightness".
  ///
  /// Defaults to "brightness".
  String get textBrightness =>
      getTranslatedString("image_picker_image_edit_brightness", "brightness");

  /// Get localized text for label "image_picker_image_edit_saturation".
  ///
  /// Defaults to "saturation".
  String get textSaturation =>
      getTranslatedString("image_picker_image_edit_saturation", "saturation");

  /// Get localized text for label "image_picker_ocr".
  ///
  /// Defaults to "OCR".
  String get textOCR => getTranslatedString("image_picker_ocr", "OCR");

  /// Get localized text for label "image_picker_request_permission".
  ///
  /// Defaults to "Request Permission".
  String get textRequestPermission => getTranslatedString(
      "image_picker_request_permission", "Request Permission");

  /// Get localized text for label "image_picker_request_camera_permission".
  ///
  /// Defaults to "You need allow camera permission.".
  String get textRequestCameraPermission => getTranslatedString(
      "image_picker_request_camera_permission",
      "You need allow camera permission.");

  /// Get localized text for label "image_picker_request_gallery_permission".
  ///
  /// Defaults to "You need allow photo gallery permission.".
  String get textRequestGalleryPermission => getTranslatedString(
      "image_picker_request_gallery_permission",
      "You need allow photo gallery permission.");

  /// Translate string by translateFunc.
  String getTranslatedString(String name, String defaultValue) {
    return translateFunc.call(name, defaultValue);
  }
}
